app/components/primer/alpha/action_list_item.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+module Primer
+  module Alpha
+    class ActionListItem < Primer::Component
+      status :alpha
+
+      DEFAULT_SIZE = :small
+      SIZE_MAPPINGS = {
+        DEFAULT_SIZE => nil,
+        :medium => "ActionList-content--sizeMedium",
+        :large => "ActionList-content--sizeLarge"
+      }.freeze
+      SIZE_OPTIONS = SIZE_MAPPINGS.keys.freeze
+
+      DEFAULT_DESCRIPTION_VARIANT = :block
+      DESCRIPTION_VARIANT_MAPPINGS = {
+        :inline => "ActionList-item-descriptionWrap--inline",
+        DEFAULT_DESCRIPTION_VARIANT => "ActionList-item-descriptionWrap--block"
+      }.freeze
+      DESCRIPTION_VARIANT_OPTIONS = DESCRIPTION_VARIANT_MAPPINGS.keys.freeze
+
+      DEFAULT_VARIANT = :default
+      VARIANT_MAPPINGS = {
+        DEFAULT_VARIANT => nil,
+        :danger => "ActionList-item--danger"
+      }.freeze
+      VARIANT_OPTIONS = VARIANT_MAPPINGS.keys.freeze
+
+      DEFAULT_SELECT_MODE = :none
+      SELECT_MODE_ICON_MAPPINGS = {
+        DEFAULT_SELECT_MODE => nil,
+        :single => "ActionList-item--singleSelectCheckmark",
+        :multiple => "ActionList-item-multiSelectIcon"
+      }.freeze
+      SELECT_MODE_OPTIONS = SELECT_MODE_ICON_MAPPINGS.keys.freeze
+
+      renders_one :description
+
+      renders_one :leading_visual, types: {
```

Comment:
I'd like to offer something similar for both `leading_action` and `trailing_action` while still configuring support for the two checked/selected states. Is that doable?

Reply:
I think so. What sorts of actions did you have in mind?

Reply:
I added this as WIP 😄